### PR TITLE
feat: add medium filtering to Artist Insights Auction Results

### PR DESCRIPTION
**Pull request by MounirDhahri**
The type of this PR is: **Feature**

This PR resolves [CX-741]

### Description
- Add medium/categories filtering to Artist Insights Auction Results
- PS: The force push is because of the rebase to master (I started this branch from the react-navigation refactor branch)

https://user-images.githubusercontent.com/11945712/104579128-dfb31400-565b-11eb-8036-1da886bb6bc2.mov

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

[CX-434]: https://artsyproduct.atlassian.net/browse/CX-434
[CX-741]: https://artsyproduct.atlassian.net/browse/CX-741

### CODE REVIEW

File: src/lib/Components/ArtworkFilterOptions/MultiSelectCheckOption.tsx [hunk 1]
```diff
@@ -39,7 +39,7 @@ export const MultiSelectCheckOptionScreen: React.FC<MultiSelectOptionScreenProps
       <FancyModalHeader onLeftButtonPress={handleBackNavigation}>{filterHeaderText}</FancyModalHeader>
       <Flex mb={120}>
         <FlatList
-          initialNumToRender={4}
+          initialNumToRender={10}
```
Comment (MounirDhahri) [inline src/lib/Components/ArtworkFilterOptions/MultiSelectCheckOption.tsx:42]: I adjusted this to render enough items for the viewable area of the screen

**Comment by admbtlr** [comment]
I find that `(1)` on the button totally confusing. To me it implies that there is one result 🤷‍♂️

File: src/lib/Components/ArtworkFilterOptions/MultiSelectCheckOption.tsx (hunk 1 above)
Comment (brainbicycle) [reply src/lib/Components/ArtworkFilterOptions/MultiSelectCheckOption.tsx:42]: This prop has bitten me so many times now 🤦

File: src/lib/Components/ArtworkFilterOptions/CategoriesOptions.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+import { FilterDisplayName, FilterParamName } from "lib/utils/ArtworkFilter/FilterArtworksHelpers"
+import { xor } from "lodash"
+import React, { useContext, useState } from "react"
+import { FilterModalNavigationStack } from "../FilterModal"
+import { MultiSelectCheckOptionScreen } from "./MultiSelectCheckOption"
+
+interface ArtistIDsArtworksOptionsScreenProps
+  extends StackScreenProps<FilterModalNavigationStack, "CategoriesOptionsScreen"> {}
+
+export const CATEGORIES_OPTIONS: FilterData[] = [
```
Comment (brainbicycle) [inline src/lib/Components/ArtworkFilterOptions/CategoriesOptions.tsx:12]: **question:** Are there no aggregate filter options for these auction results? Ideally these would be dynamic based on auction results returned from metaphysics

Comment (MounirDhahri) [reply src/lib/Components/ArtworkFilterOptions/CategoriesOptions.tsx:12]: unfortunately not, I had to use the same query that we are using in force

**Comment by MounirDhahri** [comment]
@admbtlr in the auction screen we used to show the number of available results on top of the artworks list and it helped making it easier to know that the number shown in the button is for applied filters. However in this screen we are not showing it. We can include it maybe in the future to offer a better UX

File: src/lib/Components/ArtworkFilterOptions/CategoriesOptions.tsx (hunk 2 above)
Comment (brainbicycle) [reply src/lib/Components/ArtworkFilterOptions/CategoriesOptions.tsx:12]: Got it 👍, might be worth a spike later on to see if it is possible to add this but for now looks good, merge when ready!

**Review by brainbicycle — APPROVED** [review]

**Comment by MounirDhahri** [comment]
#squashongreen

**Pull request merged**